packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -43,11 +43,11 @@ export const isDataStoreAliasMessage = (
 
 export const channelToDataStore = (
 	fluidDataStoreChannel: IFluidDataStoreChannel,
-	internalId: string,
 	runtime: ContainerRuntime,
 	datastores: DataStores,
 	logger: ITelemetryLoggerExt,
-): IDataStore => new DataStore(fluidDataStoreChannel, internalId, runtime, datastores, logger);
+): IDataStore =>
+	new DataStore(fluidDataStoreChannel, fluidDataStoreChannel.id, runtime, datastores, logger);
```

Comment:
You don't need to pass in `fluidDataStoreChannel.id`. DataStore can get it from `fluidDataStoreChannel`.

Reply:
Yes, will make this change, but @andre4i - FYI with or without proposed by Navin change, I think we will run into conflict between our changes. I think you removed ID from channel, and I'm using it, as I have no other way to get it - the caller (ContainerRuntime) does not know it (it calls into channelToDataStore), as I assign it deep inside of DataStores implementation, depending on various conditions (i.e. it's not as simple as grab uuid).